### PR TITLE
Fixed requests contrib to raise original exceptions

### PR DESCRIPTION
**Pull request by victoraugustolls**
Updates requests contrib to raise original exceptions, and updated tests.

Thanks @isra17 for noticing!

### CODE REVIEW

**Comment by victoraugustolls** [comment]
With this and #755 I'm hopping we can release a new version?
@reyang @c24t

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
 # Changelog
 
 ## Unreleased
-- Updated `django`, `flask`, `httplib`, `requests` and `pyramid` modules.
+- Updated `django`, `flask`, `httplib`, `requests` and `pyramid` modules
   ([#755](https://github.com/census-instrumentation/opencensus-python/pull/755))
 - Added `http code` to `grpc code` status code mapping on `utils`
   ([#746](https://github.com/census-instrumentation/opencensus-python/pull/746))
+- Updated `requests` module
+  ([#771](https://github.com/census-instrumentation/opencensus-python/pull/771))
```
Comment (lzchen) [inline CHANGELOG.md]: Do we need to update the CHANGELOG for the core sdk?

Comment (victoraugustolls) [reply CHANGELOG.md]: Is just for tracking for the next release, so we don’t forget to update some module. I remember someone suggested this recently, but I can’t remember in which PR. It does not need to go into the release changelog (in my opinion), but this way we know what needs to be updated !

Comment (lzchen) [reply CHANGELOG.md]: I see. As long as we are consistent, I feel like it should be fine. So far it seems like we only update the changelog if there are changes in the actual package but your argument for it being a "release reminder" is valid :).

**Comment by reyang** [comment]
@songy23 Sure, we'll take care of it.

**Pull request merged**